geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/membership/gms/locator/GMSLocatorRecoveryJUnitTest.java

```diff
@@ -194,24 +195,20 @@ public void testRecoverFromOther() throws Exception {
   }
 
   @Test
-  public void testViewFileFoundWhenUserDirModified() throws Exception {
+  public void testViewFileNotFound() throws Exception {
     NetView view = new NetView();
     populateStateFile(this.tempStateFile, GMSLocator.LOCATOR_FILE_STAMP, Version.CURRENT_ORDINAL,
         view);
+    assertTrue(this.tempStateFile.exists());
 
-    String userDir = System.getProperty("user.dir");
-    try {
-      File dir = new File("testViewFileFoundWhenUserDirModified");
-      dir.mkdir();
-      System.setProperty("user.dir", dir.getAbsolutePath());
-      File viewFileInNewDirectory = new File(tempStateFile.getName());
-      // GEODE-4180 - file in parent dir still seen with relative path
-      assertTrue(viewFileInNewDirectory.exists());
-      File locatorViewFile = locator.setViewFile(viewFileInNewDirectory);
-      assertFalse(locator.recoverFromFile(locatorViewFile));
-    } finally {
-      System.setProperty("user.dir", userDir);
-    }
+    File dir = new File("testViewFileFoundWhenUserDirModified");
+    dir.mkdir();
+    File viewFileInNewDirectory = new File(dir, tempStateFile.getName());
+
+    assertFalse(viewFileInNewDirectory.exists());
+    File locatorViewFile = locator.setViewFile(viewFileInNewDirectory);
+    assertFalse(locator.recoverFromFile(locatorViewFile));
+    FileUtils.deleteQuietly(dir);
```

Comment:
I recommend moving the dir variable to be declared for the class and then invoke deleteQuietly within tearDown(). If the test fails, it'll leave that dir around.